serveradmin/api/api.py

```diff
@@ -0,0 +1,37 @@
+import hashlib
+from datetime import timedelta
+
+from django.utils import timezone
+
+from serveradmin.api.decorators import api_function
+from serveradmin.api.models import Lock
+
+
+@api_function(group='api')
+def lock(identifier, seconds=60):
+    """Mark identifier as in-use for n seconds
+
+    Allow clients to mark resources with an identifier as in-use so that
+    distributed clients can check whatever something is in use or not.
+
+    :param identifier: A unique identifier (e.g. 10.0.0.1)
+    :param seconds: seconds until the lock expires
+
+    :return: True on success or seconds left if already in use
+    """
+
+    # Use hash sum because this has a constant length
+    hash_sum = hashlib.sha1(str(identifier).encode()).hexdigest()
+
+    # Remove expired logs first
+    Lock.objects.filter(until__lt=timezone.now()).delete()
```

Comment:
I might guess that the locking can be crucial in timing, therefore it might be rather interesting to put that into a command and calling it with a cron job here and then to slim down the view function.

Reply:
I thought about this too in the first place but decided against it. 

We don't use the API yet executing this query takes at most a few milli seconds (for a few hundred rows). I think implementing a management command to run it as a cron job would be overkill here and more error prone (people can forget to setup the cron) and this can be done easily if it turns out to be a problem.

Reply:
Fair enough